### ISSUE TITLE
Feature Request - Visualization options

### ISSUE THREAD

**Issue by yasarshaikh** [forcedotcom/sfdx-scanner#170]
Like for PMD, there is some predefined HTML. Can we get some predefined options to have great visualization tool, that any stakeholder can understand? As of now, SonarQube has a pretty great dashboard. 

Also, another option can be show report in Salesforce itself.
Since we have LWC and now some of the native(salesforce) support, can we have a report in Salesforce itself? E.g. Optimizer generates a report and shows org itself.

**Comment by jbartolotta-sfdc** [comment]
@yasarshaikh - Thanks for your request. We are currently working on an option to output an HTML formatted report, I will let you know when it is available. 

Connecting the data to a Salesforce org is something that we have discussed, but is not currently scheduled.

Does your use case involve scanning the code as it resides in the org, or scanning a repo that exists on a client machine and importing the results into an org that may or may not have the same set of metadata as the repo?

**Comment by yasarshaikh** [comment]
As of now, I've source-driven part only, so all metadata is in repo. However I was just wondering if I need to share the report to some of the stakeholders(who don't code/not technically familiar), so for them I can use either Salesforce or some PDFs. 

However having said that, it'd be great if HTML has more meaningful data in layman's terms & technical terms. E.g. prima facie it's a layman's term and for those who want to dig deeper, can have technical details.

**Comment by jbartolotta-sfdc** [comment]
@yasarshaikh I have attached a sample HTML report that was run against some of our test fixtures. It has an extra `.rename.from.txt` extension because Github doesn't allow html attachments, you will need to remove this. 

Can you let me know if this satisfies your use case? Thanks.

[sample-report.html.rename.from.txt](https://github.com/forcedotcom/sfdx-scanner/files/5138138/sample-report.html.rename.from.txt)

**Comment by yasarshaikh** [comment]
Hi @jbartolotta-sfdc  - This report looks good, however I'd like to see below options - 

The pie chart - Is it possible to have 3 pie chart : Overall a big horizontal , and below 2/3 pie chart for each technology (e.g. JS/ apex) ). 

Regarding below table I'd like to see below options - Group by severity, technology(JS/apex), files. 
When I select severity , it'll be like all files shown as collapsible[Imagine this as VS code left side file panel with collapsible options](and below actual rules and when you click to the file right side you can show the file(since its on local folder it'd easy and fast) and highlight the row with possible issues and possible fixes) - This is available in vs for pmd/eslint etc. 

I can now subgroup by technology(or I can filter independently on technology) in above groups(or I can choose file name) . WIth similar behaviors. 

Also can we show some percentage in top in a graphical way  - Percentage with severity way (imagine kind of tiles we show in salesforce dashboard), coverage percentage(not sure if its on your roadmap), percentage with tech way, and number of lines scanned per technology. (10K apex, 20K javascript). 

Also curious to know 2 things from report - 
1.0 to 5.0 what it is(below each table).  - what it is?
And in table what is end line and end column .

**Comment by jbartolotta-sfdc** [comment]
@yasarshaikh - Thanks for the valuable feedback. We will look into incorporating what we can. We will also work on ways of importing the output into an org easier, this will allow us to leverage tools that already exist for more complex scenarios.

**Answers to your questions**
`1.0 to 5.0 what it is(below each table). - what it is?`
Each file has a separate bar chart that shows the distribution by severity, i.e. a file with 1 sev1, 3 sev2's, and 4 sev3's would have 3 colored bars. It is currently a little confusing because the origin does not always start at 0, which causes some of the bars to not display. I will work on making this clearer.

`And in table what is end line and end column`
These are used for errors where a violation covers multiple lines. This is most relevant when a violation applies to a whole class or method. The end line would be the closure of the class or method. One specific example is missing ApexDoc for a class. The start line is the beginning of the class, then end line is its closing bracket.

**Comment by jbartolotta-sfdc** [comment]
@yasarshaikh - version 2.1 of the plugin contains a new `html` output option. You can upgrade the plugin via `sfdx plugins:update`. I am going to leave this issue open because the output is not finalized and we plan to make some enhancements to the report.

**Comment by rsoesemann** [comment]
@yasarshaikh You should definitely have at [Codacy.com](https://codacy.com). It's like the sfdx-scanner but as a hosted graphical service.

**Comment by yasarshaikh** [comment]
> @yasarshaikh You should definitely have at [Codacy.com](https://codacy.com). It's like the sfdx-scanner but as a hosted graphical service.

I have sonar server implemented. However its paid so I'm looking kind of same behaviors here, so that I can convince stakeholders with showing comparisons and ask them to remove that. It'll be cost effective and developer friendly solution then :)

**Comment by johnbelosf** [comment]
We added an html output option in 2020 so I'm closing this issue - please open another issue if you have more specific feedback you'd like to share. Thank you!
